src/Dialogs/StripeDialog.vala

```diff
@@ -360,41 +367,44 @@ public class AppCenter.Widgets.StripeDialog : Gtk.Dialog {
 
             var data = get_stripe_data (stripe_key, email_entry.text, (amount * 100).to_string (), card_number_entry.text, card_expiration_entry.text[0:2], year, card_cvc_entry.text);
             debug ("Stripe data:%s", data);
-            var error = false;
+            string error = null;
             try {
                 var parser = new Json.Parser ();
                 parser.load_from_data (data);
                 var root_object = parser.get_root ().get_object ();
-                var token_id = root_object.get_string_member ("id");
-
-                string? houston_data = null;
-                if (token_id != null) {
-                    houston_data = post_to_houston (stripe_key, app_id, token_id, email_entry.text, (amount * 100).to_string ());
-                    debug ("Houston data:%s", houston_data);
-                } else {
-                    error = true;
-                }
-
-                if (houston_data != null) {
-                    parser.load_from_data (houston_data);
-                    root_object = parser.get_root ().get_object ();
-
-                    if (root_object.has_member ("errors")) {
-                        error = true;
+                if (root_object != null && root_object.has_member ("id")) {
+                    var token_id = root_object.get_string_member ("id");
+                    string? houston_data = post_to_houston (stripe_key, app_id, token_id, email_entry.text, (amount * 100).to_string ());
+                    if (houston_data != null) {
+                        debug ("Houston data:%s", houston_data);
+                        parser.load_from_data (houston_data);
+                        root_object = parser.get_root ().get_object ();
```

Comment:
Both `get_root` and `get_object` could return `null`. Since this is already in trunk, I won't force you to check for that, but it would be nice to see this checked.